File: BungieNetApi/Models/SingleComponentResponseOfDestinyInventoryComponent.cs
using System.Runtime.Serialization;

namespace GhostSharper.Models
{
    [DataContract]
    public class SingleComponentResponseOfDestinyInventoryComponent
    {
        /// <summary>
        /// A list of minimal information for items in an inventory: be it a character's inventory, or a Profile's inventory. (Note that the Vault is a collection of inventory buckets in the Profile's inventory)
        /// </summary>
        /// <summary>
        /// Inventory Items returned here are in a flat list, but importantly they have a bucketHash property that indicates the specific inventory bucket that is holding them. These buckets constitute things like the separate sections of the Vault, the user's inventory slots, etc. See DestinyInventoryBucketDefinition for more info.
        /// </summary>
        [DataMember(Name = "data", EmitDefaultValue = false)]
        public DestinyInventoryComponent Data { get; set; }

        [DataMember(Name = "privacy", EmitDefaultValue = false)]
        public ComponentPrivacySetting Privacy { get; set; }

        /// <summary>
        /// If true, this component is disabled.
        /// </summary>
        [DataMember(Name = "disabled", EmitDefaultValue = false)]
        public bool Disabled { get; set; }


        public override bool Equals(object input)
        {
            return this.Equals(input as SingleComponentResponseOfDestinyInventoryComponent);
        }

        public bool Equals(SingleComponentResponseOfDestinyInventoryComponent input)
        {
            if (input == null) return false;

            return
                (
                    Data == input.Data ||
                    (Data != null && Data.Equals(input.Data))
                ) &&
                (
                    Privacy == input.Privacy ||
                    (Privacy != null && Privacy.Equals(input.Privacy))
                ) &&
                (
                    Disabled == input.Disabled ||
                    (Disabled != null && Disabled.Equals(input.Disabled))
                ) ;
        }
    }
}

File: pkg/models/SingleComponentResponseOfDestinyInventoryComponent.go
package bungieapigo

type SingleComponentResponseOfDestinyInventoryComponent struct {

	// A list of minimal information for items in an inventory: be it a character's inventory, or a
	// Profile's inventory. (Note that the Vault is a collection of inventory buckets in the
	// Profile's inventory)
	// Inventory Items returned here are in a flat list, but importantly they have a bucketHash
	// property that indicates the specific inventory bucket that is holding them. These buckets
	// constitute things like the separate sections of the Vault, the user's inventory slots, etc.
	// See DestinyInventoryBucketDefinition for more info.
	Data DestinyInventoryComponent `json:"data"`

	Privacy ComponentPrivacySetting `json:"privacy"`

	// If true, this component is disabled.
	Disabled bool `json:"disabled"`
}
